src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -79,10 +79,7 @@ public TResult[] ToArray()
                 TResult[] array = new TResult[_count];
                 if (_current != null)
```

Comment:
@jkotas, is there any efficient and built-in mechanism for checking whether a value is default(TResult)?  This check will handle reference types; I'm wondering if there's something that would more generally help with any type.